collect_app/src/test/java/org/odk/collect/android/formentry/audit/FormSaveViewModelTest.java

```diff
@@ -411,6 +411,12 @@ public void replaceRecentFileForQuestion_whenQuestionIndexHasAnswer_onRecreating
         verify(mediaUtils).deleteImageFileFromMediaProvider("blah");
     }
 
+    @Test
+    public void ignoreChanges_whenFormControllerNotSet_doesNothing() {
+        FormSaveViewModel viewModel = new FormSaveViewModel(savedStateHandle, () -> CURRENT_TIME, formSaver, mediaUtils, null);
+        viewModel.ignoreChanges(); // Checks nothing explodes
```

Comment:
Definitely could be expanded on but this is good enough to give us a harness to prevent a regression on this particular bug.

Reply:
Haven't you shaken your fist in my general direction about tests without assertions before?! 😅 

More seriously, I'm not really thinking of anything meaningful to assert and I do think it's ok.

Reply:
Haha probably. Do as I say not as I do right...

But yeah I think this test would ideally check that calls aren't made but here it felt the real focus was just to make sure nothing exploded. There is a comment 😉